src/components/structural/AppMenu.tsx

```diff
@@ -1,21 +1,177 @@
+                    {props.trailingItem && <SecondaryMenu {...props} />}
+                </div>
             </div>
         </div>
     )
 }
-
 export default AppMenu
 
 export type AppMenuProps = {
-    children?: ReactNode
+    leadingItem?: AppMenuItem
+    mainItems?: AppMenuItem[]
+    trailingItem?: AppMenuItem
+    secondaryItems?: AppMenuItem[]
+}
+
+export type AppMenuItem = AppMenuTextItem | AppMenuIconItem
+
+type AppMenuItemBase = {
+    url?: string
+    onClick?: () => void
+    testId?: string
+}
+
+export type AppMenuTextItem = {
+    title: ReactElement
+} & AppMenuItemBase
+
+export function isAppMenuTextItem(item: AppMenuItem): item is AppMenuTextItem {
+    return typeof (item as AppMenuTextItem).title === 'object'
+}
+
+export type AppMenuIconItem = {
+    icon: (props: SVGProps<SVGSVGElement>) => JSX.Element
+    iconClasses?: string
+} & AppMenuItemBase
+
+export function isAppMenuIconItem(item: AppMenuItem): item is AppMenuIconItem {
+    return typeof (item as AppMenuIconItem).icon === 'object'
+}
+
+function MenuElement(props: AppMenuItemProps): JSX.Element {
+    return (
+        <OuterAppMenuElement {...props}>
+            <InnerAppMenuElement {...props} />
+        </OuterAppMenuElement>
+    )
+}
+
+type AppMenuItemProps = {
+    fromItem?: AppMenuItem
+}
+
+type OuterAppMenuItemProps = {
+    children: ReactNode
+} & AppMenuItemProps
+
+function OuterAppMenuElement({ children, fromItem }: OuterAppMenuItemProps) {
+    if (!fromItem) return <></>
+
+    if (isAppMenuIconItem(fromItem) && fromItem.onClick != null) {
+        return (
+            <button onClick={() => fromItem.onClick?.()} data-testid={fromItem.testId || defaultTestIdElement}>
+                {children}
+            </button>
+        )
+    }
+
+    return (
+        <MenuLink to={fromItem.url || ''} data-testid={fromItem.testId || defaultTestIdElement}>
+            {children}
+        </MenuLink>
+    )
+}
+
+function InnerAppMenuElement({ fromItem }: AppMenuItemProps): JSX.Element {
+    if (fromItem && isAppMenuIconItem(fromItem)) {
+        const iconSizedClasses = `${defaultIconClasses} ${fromItem.iconClasses || 'w-7 h-7'}`
+        return <fromItem.icon className={iconSizedClasses} />
+    } else if (fromItem && isAppMenuTextItem(fromItem)) {
+        return fromItem.title
+    }
+    return <></>
+}
+
+function PrimaryMenu({ mainItems }: AppMenuProps) {
+    return (
+        <>
+            {mainItems &&
+                mainItems.map((item, index) => (
+                    <Fragment key={index}>
+                        <MenuElement fromItem={item} />
+                        {mainItems && index + 1 < mainItems.length && <span className="text-primary-300">/</span>}
+                    </Fragment>
+                ))}
+        </>
+    )
+}
+
+function SecondaryMenu({ trailingItem, secondaryItems }: AppMenuProps) {
+    return (
+        <div className="text-right">
+            <Menu as="div" className="relative block text-left">
```

Comment:
Couldn't we use the Dropdown here with adjusted styling?

Reply:
Would be desireable but would require some refactoring. It is too much effort I think